### PR TITLE
don't obfuscate error during volume create

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Comment by icecrime** [comment]
LGTM

File: daemon/create.go [hunk 1]
```diff
@@ -243,7 +243,7 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts, labels map[str
 	v, err := daemon.volumes.Create(name, driverName, opts, labels)
 	if err != nil {
 		if volumestore.IsNameConflict(err) {
-			return nil, fmt.Errorf("A volume named %s already exists. Choose a different volume name.", name)
+			return nil, errors.Wrapf(err, "a volume named %s already exists, choose a different volume name")
```
Comment (tonistiigi) [inline daemon/create.go]: missing `, name)`

Comment (vdemeester) [inline daemon/create.go]: oh you are right !

**Comment by cpuguy83** [comment]
Fixed.

**Comment by tonistiigi** [comment]
LGTM

**Comment by aaronlehmann** [comment]
I think this will lead to very ugly errors. In the simplest case, you'll get:

```
a volume named foo already exists, choose a different volume name: volume name must be unique
```

Adding `volume name must be unique` doesn't do anything to improve the error message. It just makes it redundant.

The only situation where this change would help is if `err` is something that wraps `errNameConflict`, in which case we'll have an even longer error message, containing the above plus something else.

**Comment by cpuguy83** [comment]
@aaronlehmann In the happy case when it's a true name conflict yes, it's not as succinct, however when it's some other issue with a driver it's important to be able to get this information.

This is a situation where we've found a volume in the store with the given name but ran into some error while trying to determine if the entry is stale or not.

**Comment by aaronlehmann** [comment]
Then can we omit this extra layer of wrapping? The error will always contain `volume name must be unique`.

**Comment by cpuguy83** [comment]
Done

**Comment by aaronlehmann** [comment]
LGTM

**Comment by tonistiigi** [comment]
With the current version you can just remove the extra condition as the reporting is same for all errors. To my taste is should be "failed to create volume foo: volume name must be unique".

**Comment by cpuguy83** [comment]
Fixed again.

**Comment by thaJeztah** [comment]
@cpuguy83 tests are failing

**Comment by cpuguy83** [comment]
Tests fixed.

**Review by thaJeztah — APPROVED** [review]
LGTM (if green 😅

**Comment by thaJeztah** [comment]
This a new failure?

```
17:16:51 
17:16:51 ----------------------------------------------------------------------
17:16:51 FAIL: docker_api_containers_test.go:372: DockerSuite.TestContainerAPITop
17:16:51 
17:16:51 docker_api_containers_test.go:393:
17:16:51     c.Assert(top.Processes[0][10], checker.Equals, "/bin/sh -c top")
17:16:51 ... obtained string = "top"
17:16:51 ... expected string = "/bin/sh -c top"
17:16:51 
17:16:51 
17:16:51 ----------------------------------------------------------------------
```

**Comment by cpuguy83** [comment]
Certainly unrelated. I haven't seen this before, though.

**Pull request merged**